ratis-server/src/main/java/org/apache/ratis/server/leader/LogAppenderBase.java

```diff
@@ -218,16 +221,42 @@ protected LongUnaryOperator getNextIndexForError(long newNextIndex) {
     };
   }
 
-
   @Override
-  public AppendEntriesRequestProto newAppendEntriesRequest(long callId, boolean heartbeat)
+  public AppendEntriesRequestProto newAppendEntriesRequest(long callId, boolean heartbeat) throws RaftLogIOException {
+    ReferenceCountedObject<AppendEntriesRequestProto> ref = nextAppendEntriesRequest(callId, heartbeat);
+    try {
+      return AppendEntriesRequestProto.parseFrom(ref.get().toByteString());
+    } catch (InvalidProtocolBufferException e) {
+      throw new IllegalStateException("Error copying AppendEntriesRequest.", e);
+    } finally {
+      ref.release();
+    }
+  }
```

Comment:
Since this is an implementation, we could simply throw an exception.
```java
  @Override
  public AppendEntriesRequestProto newAppendEntriesRequest(long callId, boolean heartbeat) {
    throw new UnsupportedOperationException("Use nextAppendEntriesRequest(" + callId + ", " + heartbeat +") instead.");
  }
```